src/JBDistributor.sol

```diff
@@ -87,15 +91,67 @@ abstract contract JBDistributor is IJBDistributor {
         return _snapshotAtRoundOf[token][round];
     }
 
-    /// @custom:param tokenId The ID of the token to which the vesting amount belongs.
-    /// @custom:param round The round during which the vesting began.
-    /// @custom:param token The address of the token being vested.
-    function vestingTokenAmountAtRoundOf(uint256 tokenId, uint256 round, IERC20 token)
-        external
-        view
-        returns (VestingData memory)
-    {
-        return _vestingTokenAmountAtRoundOf[tokenId][round][token];
+    /// @notice Calculate how much of the token is claimed for the given tokenId.
+    /// @param tokenId The ID of the token to calculate the token amount for.
+    /// @param token The address of the token being claimed.
+    /// @return _tokenAmount The amount of tokens that can be claimed once they have vested.
+    function claimedFor(uint256 tokenId, IERC20 token) external view returns (uint256 _tokenAmount) {
+        // Keep a refrence to the latest vested index.
+        uint256 _vestedIndex = latestVestedIndexOf[tokenId][token];                
+
+        // Keep a reference to the number of vesting rounds for the tokenId and token.
+        uint256 _numberOfVestingRounds = vestingDataOf[tokenId][token].length;
+
+        while (_vestedIndex < _numberOfVestingRounds) {
+            // Keep a reference to the vested data being iterated on.
+            VestingData memory _vesting = vestingDataOf[tokenId][token][_vestedIndex];
+
+            _tokenAmount += mulDiv(
+                _vesting.amount,
+                MAX_SHARE - _vesting.shareClaimed,
+                MAX_SHARE
+            );
+
+            unchecked {
+                ++_vestedIndex;
+            }
+        }
+    }
+
+    /// @notice Calculate how much of the token is currently ready to be collected for the given tokenId.
+    /// @param tokenId The ID of the token to calculate the token amount for.
+    /// @param token The address of the token being claimed.
+    /// @return _tokenAmount The amount of tokens that can be claimed right now.
+    function collectableFor(uint256 tokenId, IERC20 token) external view returns (uint256 _tokenAmount) {
```

Comment:
can probably also add another view that returns an array of vest structs that have amounts + dates